src/MarginTrading.SettingsService.SqlRepositories/Repositories/AssetPairsRepository.cs

```diff
@@ -76,25 +89,121 @@ public async Task<bool> TryInsertAsync(IAssetPair obj)
+                    await _log.WriteErrorAsync(nameof(AssetPairsRepository),
+                        nameof(InsertBatchAsync), $"Failed to perform batch transaction: {ex.Message}", ex);
+                    
+                    return null;
+                }
             }
         }
 
-        public async Task UpdateAsync(IAssetPair obj)
+        public async Task<IAssetPair> UpdateAsync(IAssetPair assetPair)
         {
             using (var conn = new SqlConnection(_connectionString))
             {
                 await conn.ExecuteAsync(
                     $"update {TableName} set {GetUpdateClause} where Id=@Id", 
-                    _convertService.Convert<IAssetPair, AssetPairEntity>(obj));
+                    _convertService.Convert<IAssetPair, AssetPairEntity>(assetPair));
+
+                return await conn.QuerySingleOrDefaultAsync<AssetPairEntity>(
+                    $"SELECT * FROM {TableName} WHERE Id=@id", new {assetPair.Id});
+            }
+        }
+
+        public async Task<IReadOnlyList<IAssetPair>> UpdateBatchAsync(IReadOnlyList<IAssetPair> assetPairs)
+        {
+            using (var conn = new SqlConnection(_connectionString))
+            {
+                SqlTransaction transaction = null;
+                
+                try
+                {
+                    if (conn.State != ConnectionState.Open)
+                    {
+                        await conn.OpenAsync();
+                    }
+                    
+                    transaction = conn.BeginTransaction();
+
+                    if (await conn.ExecuteScalarAsync<int>(
+                            $"SELECT COUNT(*) FROM {TableName} WITH (UPDLOCK) WHERE Id IN ({string.Join(",", assetPairs)})",
+                            new { },
+                            transaction) != assetPairs.Count)
+                    {
+                        throw new ArgumentOutOfRangeException(nameof(assetPairs), "One of asset pairs does not exist");
+                    }
+
+                    await conn.ExecuteAsync(
+                        $"update {TableName} set {GetUpdateClause} where Id=@Id",
+                        assetPairs.Select(_convertService.Convert<IAssetPair, AssetPairEntity>).ToList(),
+                        transaction);
+
+                    var inserted = await conn.QueryAsync<AssetPairEntity>(
+                        $"SELECT * FROM {TableName} WHERE Id IN ({string.Join(",", assetPairs.Select(x => x.Id))})");
+
+                    transaction.Commit();
+                    
+                    return inserted.ToList();
+                }
+                catch (Exception ex)
+                {
+                    transaction?.Rollback();
+                    await _log.WriteErrorAsync(nameof(AssetPairsRepository),
+                        nameof(InsertBatchAsync), $"Failed to perform batch transaction: {ex.Message}", ex);
+                    
+                    return null;
```

Comment:
Null value here causes errors in controller. I think we should re-throw error or mark method with [CanBeNull] attribute and handle this situation in controller.
And need to check similar places

Reply:
ItemCanBeNull attribute added, null check made in controller.